libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -517,11 +546,11 @@ private PermissionCheckRunnable obtainPermissionCheckRunnable() {
   }
 
   private void startLocation() {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-      if (ProcessLifecycleOwner.get().getLifecycle().getCurrentState().isAtLeast(Lifecycle.State.STARTED)) {
+    if (isLollipopOrHigher()) {
+      if (isAppInForeground()) {
         applicationContext.startService(obtainLocationServiceIntent());
```

Comment:
how about just catching `java.lang.IllegalStateException` here and on line 556? would that be sufficient to remedy the problem while we refactor out of `TelemetryService`?